File: src/test/java/no/nav/syfo/config/LocalApplicationConfig.java
package no.nav.syfo.config;

import no.nav.security.oidc.test.support.spring.TokenGeneratorConfiguration;
import no.nav.syfo.exception.RestTemplateErrorHandler;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.web.client.RestTemplateBuilder;
import org.springframework.context.annotation.*;
import org.springframework.http.client.ClientHttpRequestInterceptor;
import org.springframework.web.client.RestTemplate;

import static java.util.Arrays.asList;

@Configuration
@EnableAspectJAutoProxy
@Import({
        AdresseregisterConfig.class,
        FastlegeInformasjonConfig.class,
        BrukerprofilConfig.class,
        TokenGeneratorConfiguration.class,
})
@Profile("local")
public class LocalApplicationConfig {

    @Bean(name = "Oidc")
    @Primary
    public RestTemplate restTemplate(ClientHttpRequestInterceptor... interceptors) {
        RestTemplate template = new RestTemplate();
        template.setInterceptors(asList(interceptors));
        template.setErrorHandler(new RestTemplateErrorHandler());
        return template;
    }

    @Bean(name = "BasicAuth")
    public RestTemplate basicAuthRestTemplate(@Value("${srv.username}") String username,
                                              @Value("${srv.password}") String password) {
        return new RestTemplateBuilder()
                .basicAuthentication(username, password)
                .build();
    }

}

File: src/test/kotlin/no/nav/syfo/LocalApplicationConfig.kt
package no.nav.syfo

import no.nav.security.token.support.test.spring.TokenGeneratorConfiguration
import no.nav.syfo.api.exception.RestTemplateErrorHandler
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.web.client.RestTemplateBuilder
import org.springframework.context.annotation.*
import org.springframework.http.client.ClientHttpRequestInterceptor
import org.springframework.web.client.RestTemplate
import java.util.*

@Configuration
@EnableAspectJAutoProxy
@Import(
    TokenGeneratorConfiguration::class
)
@Profile("local")
class LocalApplicationConfig {
    @Bean(name = ["Oidc"])
    @Primary
    fun restTemplate(vararg interceptors: ClientHttpRequestInterceptor): RestTemplate {
        val template = RestTemplate()
        template.interceptors = Arrays.asList(*interceptors)
        template.errorHandler = RestTemplateErrorHandler()
        return template
    }

    @Bean(name = ["BasicAuth"])
    fun basicAuthRestTemplate(
        @Value("\${srv.username}") username: String,
        @Value("\${srv.password}") password: String
    ): RestTemplate {
        return RestTemplateBuilder()
            .basicAuthentication(username, password)
            .build()
    }

    @Bean(name = ["default"])
    fun defaultRestTemplate(): RestTemplate {
        return RestTemplate()
    }
}
